src/prefect/utilities/templating.py

```diff
@@ -78,46 +96,58 @@ def apply_values(template: T, values: Dict[str, Any]) -> Union[T, Unset]:
     Returns:
         The template with the values applied
     """
-    if isinstance(template, (int, float, bool, Unset)):
+    if (
+        isinstance(template, (int, float, bool))
+        or template is NotSet
+        or template is None
```

Comment:
ha i added this in my projects PR - fwiw you can do `isinstance(template, (int, float, bool, type(None), type(NotSet))` to keep it all in one place above